File: scripts/hf_deploy.mjs
import { commit } from "@huggingface/hub";
import fs from "fs";
import path from "path";
import dotenv from "dotenv";

dotenv.config();

const TOKEN = process.env.HF_TOKEN;

async function walk(dir, fileList = []) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.git' && file !== '_debug_archive') {
                await walk(fullPath, fileList);
            }
        } else {
            const relPath = path.relative(process.cwd(), fullPath);
            if (!relPath.includes('.env') && !relPath.includes('pincodes.json') && !relPath.endsWith('.log')) {
                fileList.push({ fullPath, relPath });
            }
        }
    }
    return fileList;
}

async function deploy() {
    console.log('🚀 Starting Hugging Face Hub deployment...');
    if (!TOKEN) {
        console.error('❌ HF_TOKEN is missing in .env');
        return;
    }

    const files = await walk(process.cwd());
    const operations = [];

    for (const file of files) {
        const content = fs.readFileSync(file.fullPath);
        const relativePath = file.relPath.replace(/\\/g, '/');
        
        operations.push({
            operation: "addOrUpdate",
            path: relativePath,
            content: new Blob([content])
        });
    }

    try {
        console.log(`📦 Committing ${operations.length} files to spaces/herbon123/Ordermanagement...`);
        const res = await commit({
            repo: { type: "space", name: "herbon123/Ordermanagement" },
            credentials: { accessToken: TOKEN },
            operations: operations,
            title: "Supercharged: Add backend pagination & lean optimization"
        });
        console.log('🏁 Deployment complete! The Space is now building.', res.commit.id);
    } catch (e) {
        console.error('❌ Abort due to error:', e.message);
    }
}

deploy();

File: scripts/hf_deploy.py
import os
import requests

TOKEN = os.environ.get("HF_TOKEN")
REPO = "herbon123/Ordermanagement"
BASE_URL = f"https://huggingface.co/api/spaces/{REPO}/upload/main/"

def upload_folder(folder_path):
    print(f"🚀 Starting API upload for: {folder_path}")
    for root, dirs, files in os.walk(folder_path):
        # Skip internal git stuff
        if ".git" in root:
            continue
        
        for file in files:
            full_path = os.path.join(root, file)
            # Calculate relative path for the remote
            rel_path = os.path.relpath(full_path, folder_path).replace("\\", "/")
            
            # Skip noise
            if rel_path == "push_error.txt" or ".env" in rel_path:
                continue

            print(f"📤 Uploading {rel_path}...")
            url = BASE_URL + rel_path
            try:
                with open(full_path, "rb") as f:
                    res = requests.post(
                        url, 
                        headers={"Authorization": f"Bearer {TOKEN}"}, 
                        data=f
                    )
                    if res.status_code >= 200 and res.status_code < 300:
                        print(f"✅ Done")
                    else:
                        print(f"❌ Error {res.status_code}: {res.text}")
            except Exception as e:
                print(f"💥 Fatal error: {str(e)}")

# Ensure requests is installed
import subprocess
subprocess.check_call(["pip", "install", "requests"])

# Run upload
upload_folder("hf_deploy")
print("🏁 All done!")
